### PR TITLE
Update pillow to 10.0.0 and fix text size calculation in text_filter

### PR DESCRIPTION
**Pull request by irthomasthomas**
A quick fix for #353

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ readme = "README.md"
 [tool.poetry.dependencies]
 python = ">=3.8,<3.12"
 streamdeck = "^0.9.3"
-pillow = "^9.4.0"
+pillow = "^10.0.0"
```
Comment (hugovk) [inline pyproject.toml:12]: I don't think you necessarily need to require 10+, textlength and textbbox were added in 8.0.0: 

* https://pillow.readthedocs.io/en/stable/releasenotes/8.0.0.html#api-additions
* https://github.com/python-pillow/Pillow/pull/4959

```suggestion
pillow = "^9.4.0"
```

**Comment by irthomasthomas** [comment]
> would be nice to see this merged. But its me or this project feels somewhat abandoned?
> 
> @timothycrosley last commit on Github was on March 13 2023. The last commit on this project dates from April 30.
> 
> Pull requests and issues are piling. Hard to tell the direction this is going and quite sad since its probably the best tool out there for streamdeck on linux

I'm actually worried, now. I just checked his Twitter, and his last tweet was a new year's resolution to make one commit every day, but he hasn't made any commits since March.  Hopefully, he's just taking a break from online. If not, we will have to fork it. @4s3ti @hugovk @abcminiuser @lhayhurst

**Comment by Danie10** [comment]
I've sent an e-mail now to Timothy's GMail address asking if he will respond here to let us know.

**Comment by Danie10** [comment]
No response yet to the mail I sent to @timothycrosley 46 hours ago.

**Comment by Data-007** [comment]
@Danie10  Frankly I fear that more is need. I emailed you about @dodgyrabbit 
Maybe its time to take it on. I do understand that Pieter is very busy.

I would love to see some improvements,  #208 being one prominent  one im most
interested in I think from what Pieter has typed that he has a ruff working copy.

Plz look over issues tag #208 in issues.
Thanks

**Comment by Danie10** [comment]
@Data-007 I received no e-mail tho? Yes, I see @dodgyrabbit has two stream deck projects but both updated more than a year ago. It would be great to be able to all move to an agreed fork, and also be able to do an import of what we have already set up if possible (easy migration path).

**Comment by Data-007** [comment]
Hi
I think I got mixed up with who I emailed or not. But I agree in part, I think that it is important to do this right.
To my knowledge it has all-way's been Pieter that's done most of it.  It to me does not feel  right to just fork and
leave without....

Would be nice to see some improvements, thee #208, plugins, the improvements that everyone wants.

**Comment by Data-007** [comment]
@Danie10 @
> @Data-007 I received no e-mail tho? Yes, I see @dodgyrabbit has two stream deck projects but both updated more than >a year ago. It would be great to be able to all move to an agreed fork, and also be able to do an import of what we have >already set up if possible (easy migration path).

Ok Im a novice at GitHub however I have forked the Repo, if anyone wants to help/help with setup/submit stuff.
Have a look @Data-007  public repo it's a carbon copy, wort's an all.
Data.

**Comment by Danie10** [comment]
Thanks @Data-007 but that was also why I did not fork it myself. I'm not really experienced at managing multiple commits and branches on Github. It requires the active participation by the key guys who would do the coding (those so far submitting workflow approvals here), and maybe one of them wants to take the lead via their project. But let's hear what they have to say. I'm hoping that we all move to one new fork.

**Comment by 4s3ti** [comment]
ok ... sorry ... but the way this is going worries me ... 

first ... let's not take rushed steps. 

@Data-007  
> Ok Im a novice at GitHub however I have forked the Repo, if anyone wants to help/help with setup/submit stuff. 

this doesn't bring me much confidence ... If we are taking over a project because the original owner vanished, then in a project like this needs someone that is more comfortable around, but on the other hand,I am perfectly fine with that (no one was born with all the knowledge and swimming with the sharks is a perfect way to do so!) as long as there are multiple maintainers and open comms between all of them.  needs to have multiple maintainers ... to avoid again the BUS Factor.

Whoever is in charge also should account that there are probably some other projects / packages relying on this, I can count at least the Arch AUR package manager. 

I suggest to get a list of volunteers to Fork and maintain the project, then find a place where we all can talk and agree with the path forward and then take action. 

Even tho I know some python I have very little experience with it, would need for sure much more help in code related tasks.  but I am very comfortable with everything else and I am ok into taking care of almost everything related with the project, from seting up the repo, to testing and release pipelines.

This being said, regardless of where this is forked into, I am here to help with anything. I have been wanting to contribute to this project in a while so the opportunity presented itself!

**Comment by 4s3ti** [comment]
Ok ... let's take this discussion here: https://github.com/timothycrosley/streamdeck-ui/issues/357 

as we are driving really away from the topic of this pull request.

**Comment by 4s3ti** [comment]
@irthomasthomas could you pls make this PR in the new repository?

https://github.com/streamdeck-linux-gui/streamdeck-linux-gui 

if not asking too much please try to follow the new contribution rules to make things smoother :) 

Hopefully the AUR package will also piggy back on this and we will get it there :) 

thanks a lot.

**Comment by irthomasthomas** [comment]
> 

Thanks! I'll get on it, now.

**Pull request closed without merging**